app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/MoreDialog/MoreDialog.tsx

```diff
@@ -16,33 +15,39 @@ import { DynamicRowOptions } from "./DynamicRowOptions";
 import { TextFieldOptions } from "./TextFieldOptions";
 import { CharacterLimitOptions } from "./CharacterLimitOptions";
 import { useRefsContext } from "@formBuilder/[id]/edit/components/RefsContext";
+import { FormElement } from "@lib/types";
 
 export const MoreDialog = () => {
-  const { elements, updateField, setChangeKey } = useTemplateStore((s) => ({
+  const { elements, updateField, setChangeKey, getElementById } = useTemplateStore((s) => ({
     lang: s.lang,
     updateField: s.updateField,
     elements: s.form.elements,
     getFocusInput: s.getFocusInput,
     setChangeKey: s.setChangeKey,
+    getElementById: s.getElementById,
   }));
 
-  const [item, setItem] = React.useState<FormElementWithIndex | null>(null);
+  const [item, setItem] = React.useState<FormElement | undefined>(undefined);
   const [isOpen, setIsOpen] = React.useState(false);
   const { Event } = useCustomEvent();
   const dialog = useDialogRef();
   const { refs } = useRefsContext();
   const { t } = useTranslation("form-builder");
 
   type MoreDialogEventDetails = {
-    item: FormElementWithIndex;
+    itemId: number;
   };
 
-  const handleOpenDialog = useCallback((detail: MoreDialogEventDetails) => {
-    if (detail) {
-      setItem(detail.item);
-      setIsOpen(true);
-    }
-  }, []);
+  const handleOpenDialog = useCallback(
+    (detail: MoreDialogEventDetails) => {
+      if (detail) {
+        const freshItem = getElementById(detail.itemId);
```

Comment:
Thinking we should rename this to something that doesn't sound like it's the build in dom method

getFormElementById or getTemplateElementById or getItemById